gcloud-java-dns/src/test/java/com/google/cloud/dns/DnsBatchTest.java

```diff
@@ -0,0 +1,632 @@
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    batchMock.addGetZone(EasyMock.eq(ZONE_NAME), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.expect(optionsMock.service()).andReturn(dns);
+    EasyMock.replay(dns, batchMock, optionsMock);
+    DnsBatchResult<Zone> batchResult = dnsBatch.getZone(ZONE_NAME, ZONE_FIELDS);
+    assertNotNull(callback.getValue());
+    String selector = (String) capturedOptions.getValue().get(ZONE_FIELDS.rpcOption());
+    assertTrue(selector.contains(Dns.ZoneField.CREATION_TIME.selector()));
+    assertTrue(selector.contains(Dns.ZoneField.NAME.selector()));
+    RpcBatch.Callback<ManagedZone> capturedCallback = callback.getValue();
+    capturedCallback.onSuccess(ZONE_INFO.toPb());
+    assertEquals(ZONE_INFO.toPb(), batchResult.get().toPb());
+  }
+
+  @Test
+  public void testDeleteZone() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<Void>> callback = Capture.newInstance();
+    batchMock.addDeleteZone(EasyMock.eq(ZONE_NAME), EasyMock.capture(callback));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<Boolean> batchResult = dnsBatch.deleteZone(ZONE_NAME);
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
+    } catch (IllegalStateException ex) {
+      // expected
+    }
+    // testing error here, success is tested with options
+    RpcBatch.Callback<Void> capturedCallback = callback.getValue();
+    capturedCallback.onFailure(GOOGLE_JSON_ERROR);
+    try {
+      batchResult.get();
+      fail("Should throw a DnsException on error.");
+    } catch (DnsException ex) {
+      // expected
+      assertFalse(ex.idempotent());
+    }
+  }
+
+  @Test
+  public void testDeleteZoneOnSuccess() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<Void>> callback = Capture.newInstance();
+    batchMock.addDeleteZone(EasyMock.eq(ZONE_NAME), EasyMock.capture(callback));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<Boolean> batchResult = dnsBatch.deleteZone(ZONE_NAME);
+    assertNotNull(callback.getValue());
+    RpcBatch.Callback<Void> capturedCallback = callback.getValue();
+    Void result = null;
+    capturedCallback.onSuccess(result);
+    assertTrue(batchResult.get());
+  }
+
+  @Test
+  public void testGetProject() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<Project>> callback = Capture.newInstance();
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    batchMock.addGetProject(EasyMock.capture(callback), EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<ProjectInfo> batchResult = dnsBatch.getProject();
+    assertEquals(0, capturedOptions.getValue().size());
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2428842103/issues/source?bid=3123103&fileBranchId=3240468#l354)